### PR TITLE
Fix LanguageResources data format

### LINKED ISSUE #48: LanguageResources not following the expected json structure
## The problem
The current `LanguageResources`  deserializes the JSON as a `string[]`, meaning our language files should follow this structure:
One file per key, with the following structure. 
```json
[
    "phrase1", 
    "prase2" 
]
```
## Expected behavior
But it should be one file per language, containing pairs of key, as mentioned [here](https://github.com/discord-bot-tutorial/Miunie/pull/41#issuecomment-457834989):
```json
{
    "SomeKey" : [ "Some Value!" ],
    "SomeKey2" : [ "Some Value2", "Some Value3" ]
}
```
## Suggested changes
- One file per language, with multiple `key : values` pairs, where `values` is an array. In C#, this is a `Dictionary<string, string[]>`.
- The `GetPhrase(phraseKey)` method, there is only one value, it returns that value, but if there are multiple, returns you a random one. 

- I also suggest to add an overload to support multiple languages, wich would be the entity key (in this case, the file name).
- As reminder, every new feature / change **must** have its own Unit Tests.

### PR DESCRIPTION
**Pull request by Charly6596**
## Related Issue

Fixes #48 

## Changes
- Modified and refactored `LanguageResources` to expect a `Dictionary<string, string[]>` from the data storage, where the `string` is the key, and the `string[]` are the phrases. That object's key indicates the language.
- Refactored the tests to support the expected data structure
- Added new methods to indicate the language, `GetPhrase(key, lang)`, and `GetTranslatedFormatted(key, lang, params)`.
- Written tests to verify the new methods to support multiple language works. 
## Expected Feedback
:warning: `LanguageResources` doesn't verify if the language exists, perhaps we should return a default language phrase if doesn't exists the specified language?

### CODE REVIEW

File: src/Miunie.Core/Services/ILanguageResources.cs [hunk 1]
```diff
@@ -3,6 +3,11 @@ namespace Miunie.Core
     public interface ILanguageResources
     {
         string GetPhrase(string key);
+        string GetPhrase(string key, string lang);
         string GetFormatted(string key, params object[] objs);
+        string GetTranslatedFormatted(
+                                string key,
+                                string lang,
+                                params object[] objs);
```
[the remark this answers is not in the document]
Comment (Charly6596) [reply src/Miunie.Core/Services/ILanguageResources.cs]: Thanks for taking the time to explain that, makes a lot of sense (: I kinda dislike the way to set the language tho, for not being thread safe, but couldn't figure out a better solution

**Comment by Charly6596** [comment]
## Changes
- Modified `ILanguageResources` to be Transient instead of a Singleton.
- Removed the `GetPhrase` overloads, keeping `string GetPhrase(string key, params object[] objs)` and 
`void SetLanguage(string langKey)` to set the language.
- Refactored tests.

Here's how and where the .json file should be at https://github.com/discord-bot-tutorial/Miunie/issues/50#issuecomment-459415370
> NOTE: Modified the path to Resources/**Lang**/Phrases{LangKey} due to `LangResources` being a bit redundant

File: src/Miunie.Core/Services/LanguageResources.cs [hunk 2]
```diff
@@ -9,37 +10,43 @@ public class LanguageResources : ILanguageResources
...
-        public string GetFormatted(string key, params object[] objs)
+        private string GetLangKey(string langKey)
+            => String.Format(_langKeyFormat, langKey);
+
+        private string PickRandom(string[] collection)
         {
-            var phrase = GetPhrase(key);
-            return String.Format(phrase, objs);
+            if (collection is null) { return String.Empty; }
+            if (collection.Length == 1) { return collection[0]; }
```
Comment (DraxCodes) [inline src/Miunie.Core/Services/LanguageResources.cs]: This seems redudent as `_rand.Next(collection.Length);` will always return the first and only index if only 1 item is in the collection.

**Comment by petrspelos** [comment]
@Charly6596 please update your branch. :open_hands:

**Comment by petrspelos** [comment]
I'm looking at this, don't worry. Just taking my time. :blush: :tada: 
## Pull request check progress
- [ ] Stylistic changes
- [ ] Tests ran locally
- [ ] Bot ran locally

**Comment by Charly6596** [comment]
Nice job reworking that @petrspelos :+1: looks better.

**Pull request merged**